backend/src/main/scala/cromwell/backend/BackendWorkflowInitializationActor.scala

```diff
@@ -31,31 +40,86 @@ object BackendWorkflowInitializationActor {
 trait BackendWorkflowInitializationActor extends BackendLifecycleActor with ActorLogging {
 
   def receive: Receive = LoggingReceive {
-    case Initialize    => performActionThenRespond(initSequence, onFailure = InitializationFailed)
-    case AbortWorkflow => performActionThenRespond(abortInitialization, onFailure = BackendWorkflowAbortFailedResponse)
+    case Initialize => performActionThenRespond(initSequence(), onFailure = InitializationFailed)
+    case AbortWorkflow => performActionThenRespond(abortInitialization(), onFailure = BackendWorkflowAbortFailedResponse)
   }
 
   /**
     * Our predefined sequence to run during preStart
     */
-  final def initSequence = for {
-    _ <- validate
-    _ <- beforeAll
-  } yield InitializationSuccess
+  final def initSequence() = {
+    for {
+      _ <- validate()
+      _ <- beforeAll()
+    } yield InitializationSuccess
+  }
 
   /**
     * Abort all initializations.
     */
-  def abortInitialization: Future[WorkflowAbortResponse]
+  def abortInitialization(): Future[WorkflowAbortResponse]
+
+  /**
+    * A call which happens before anything else runs
+    */
+  def beforeAll(): Future[Unit]
+
+  /**
+    * Validates runtime attributes for one specific call.
+    *
+    * @param runtimeAttributes Runtime Attributes with already evaluated values.
+    * @return If all entries from runtime attributes section are valid Success otherwise
+    *         Failure with the aggregation of errors.
+    */
+  def validateRuntimeAttributes(runtimeAttributes: EvaluatedRuntimeAttributes): ErrorOr[Product]
 
   /**
     * Validate that this WorkflowBackendActor can run all of the calls that it's been assigned
     */
-  def validate: Future[Unit]
+  private def validate(): Future[Unit] = {
```

Comment:
After looking at your `LocalInitializationActor`, I've reconsidered. I like this structure a lot but think we should have a some sort of `customValidation` in the `initSequence` just in case someone want some totally orthogonal custom validation of their own devising.